src/aiida/orm/nodes/data/code/portable.py

```diff
@@ -177,3 +178,31 @@ def filepath_executable(self, value: str) -> None:
             raise ValueError('The `filepath_executable` should not be absolute.')
 
         self.base.attributes.set(self._KEY_ATTRIBUTE_FILEPATH_EXECUTABLE, value)
+
+    @property
+    def filepath_files(self) -> pathlib.PurePosixPath:
+        """Return the absolute path of the directory that contains the code files.
+
+        :return: The directory of the code files required for the `filepath_executable`.
+        """
+        return pathlib.PurePosixPath(self.base.attributes.get(self._KEY_ATTRIBUTE_FILEPATH_FILES))
+
+    @filepath_files.setter
+    def filepath_files(self, value: t.Union[str, pathlib.Path]) -> None:
+        """Set the absolute path of the directory that contains the code files.
+
+        :param value: The absolute filepath of the directory of the uploaded code files.
+        """
+        type_check(value, (str, pathlib.Path))
+
+        if not pathlib.PurePosixPath(value).is_absolute():
+            raise ValueError('`filepath_files` must be an absolute path.')
+
+        if not pathlib.Path(value).exists():
+            raise ValueError('`filepath_files` must be an existing directory.')
+
+        self.base.attributes.set(self._KEY_ATTRIBUTE_FILEPATH_FILES, str(value))
+
+        if len(self.base.repository.list_object_names()) > 0:
```

Comment:
why the check first? Why not just always call erase straight away?

Reply:
Fair enough. Actually, I wasn't sure about the `erase` anyway, as I didn't want anything to be deleted that the user might not want to be deleted (haven't really used `PortableCode` myself so far). Though, I think erasing makes sense as to not have a mess in the `repository`, with _previous_ files still being there. If one had set `filepath_files` but then modifies it, the previous `filepath_files` shouldn't be there anymore for the reason above, so yes, I think always just erasing should make sense.